packages/edit-site/src/components/page-pages/index.js

```diff
@@ -107,13 +116,26 @@ function useView( type ) {
 		[ editEntityRecord, editedViewRecord?.id ]
 	);
 
+	const setDefaultViewAndUpdateUrl = useCallback(
+		( viewToSet ) => {
+			if ( viewToSet.type !== view?.type ) {
+				history.push( {
+					...params,
+					type: viewToSet.type,
+				} );
+			}
+			setView( viewToSet );
```

Comment:
In addition to updating the view object, we also update the url if the "view type" changes. Maybe a better way is to split out the type from the view state.